src/bitcoin-cli.cpp

```diff
@@ -986,8 +986,23 @@ static void ParseGetInfoResult(UniValue& result)
         RESET);
     result_string += strprintf("Version: %s\n", result["version"].getValStr());
     result_string += strprintf("Time offset (s): %s\n", result["timeoffset"].getValStr());
-    const std::string proxy = result["proxy"].getValStr();
-    result_string += strprintf("Proxy: %s\n", proxy.empty() ? "N/A" : proxy);
+
+    std::map<std::string, std::vector<std::string>> proxy_networks;
+
+    for (const UniValue& network : result["networks"].getValues()) {
+        std::string proxy = network["proxy"].getValStr();
+        if (proxy.empty()) continue;
+
+        std::string network_name = network["name"].getValStr();
+        proxy_networks[proxy].push_back(network_name);
+    }
+
+    std::vector<std::string> proxies;
+    for (const auto& [proxy, networks] : proxy_networks) {
+        proxies.push_back(strprintf("%s (%s)", proxy, Join(networks, ", ")));
+    }
+
+    result_string += strprintf("Proxies: %s\n", proxies.empty() ? "N/A" : Join(proxies, " "));
```

Comment:
a few suggestions to pick/choose/ignore

- maybe separate the proxies with a comma-space
- maybe add a comment, remove unneeded localvars, and keep the lines together to see a glance that it's one section for the proxies
- maybe `s/proxies.push_back/proxies.emplace_back/` to construct in-place
- maybe print "n/a" rather than "N/A", like line 1001 for -netinfo

```diff
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -987,22 +987,19 @@ static void ParseGetInfoResult(UniValue& result)
     result_string += strprintf("Version: %s\n", result["version"].getValStr());
     result_string += strprintf("Time offset (s): %s\n", result["timeoffset"].getValStr());
 
+    // proxies
     std::map<std::string, std::vector<std::string>> proxy_networks;
-
     for (const UniValue& network : result["networks"].getValues()) {
-        std::string proxy = network["proxy"].getValStr();
+        const std::string proxy = network["proxy"].getValStr();
         if (proxy.empty()) continue;
-
-        std::string network_name = network["name"].getValStr();
-        proxy_networks[proxy].push_back(network_name);
+        proxy_networks[proxy].push_back(network["name"].getValStr());
     }
-
     std::vector<std::string> proxies;
     for (const auto& [proxy, networks] : proxy_networks) {
-        proxies.push_back(strprintf("%s (%s)", proxy, Join(networks, ", ")));
+        proxies.emplace_back(strprintf("%s (%s)", proxy, Join(networks, ", ")));
     }
+    result_string += strprintf("Proxies: %s\n", proxies.empty() ? "n/a" : Join(proxies, ", "));
 
-    result_string += strprintf("Proxies: %s\n", proxies.empty() ? "N/A" : Join(proxies, " "));
     result_string += strprintf("Min tx relay fee rate (%s/kvB): %s\n\n", CURRENCY_UNIT, result["relayfee"].getValStr());
```

Reply:
Thanks for the suggestion! Updated the PR to adopt the suggestion in dedbc9d